src/app/AttributeAccessInterface.h

```diff
@@ -292,6 +292,7 @@ class AttributeValueEncoder
 
         mCurrentEncodingListIndex++;
         mEncodeState.mCurrentEncodingListIndex++;
+        mEncodedAtLeastOneListItem = true;
```

Comment:
Is it possible to prove that the desired behavior is happening via a unit test?

Reply:
Well, you mean apart from the unit test I had to change because it's happening?

I could try to write a targeted unit test that aims to explicitly exercise this behavior, yes.  Let me see what I can do with that.

Reply:
Done in https://github.com/project-chip/connectedhomeip/pull/28346/commits/86fc2f676bbc0dd2a288f49a558ce3e834081baa